src/mainloop.cpp

```diff
@@ -154,24 +154,48 @@ void Mainloop::route_msg(struct buffer *buf, int target_sysid, int target_compid
     bool unknown = true;
 
     for (Endpoint **e = g_endpoints; *e != nullptr; e++) {
-        if ((*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid, msg_id)) {
+        switch (
+            (*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid, msg_id)) {
+        case Endpoint::AcceptState::Accepted:
             log_debug("Endpoint [%d] accepted message %u to %d/%d from %u/%u", (*e)->fd, msg_id,
                       target_sysid, target_compid, sender_sysid, sender_compid);
             write_msg(*e, buf);
             unknown = false;
+            break;
+        case Endpoint::AcceptState::Filtered:
+            log_debug("Endpoint [%d] filtered out message %u to %d/%d from %u/%u", (*e)->fd, msg_id,
+                      target_sysid, target_compid, sender_sysid, sender_compid);
+            unknown = false;
+            break;
+        case Endpoint::AcceptState::Rejected:
+            // fall through
+        default:
+            break; // do nothing (will count as unknown)
         }
     }
 
     for (struct endpoint_entry *e = g_tcp_endpoints; e; e = e->next) {
-        if (e->endpoint->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid,
-                                    msg_id)) {
+        switch (e->endpoint->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid,
```

Comment:
ditto.  I think I had a patch to avoid this double loop and code duplication, but it's fine for now

Reply:
I will take a look at the config flow through the code as my next task (therefore all the cleanup before doing the actual changes). So the double loop might get resolved during that change as well.